php-checks/src/main/java/org/sonar/php/checks/utils/ArgumentVerifierValueContainment.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.utils;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.sonar.php.utils.collections.SetUtils;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+
+public class ArgumentVerifierValueContainment extends ArgumentMatcherValueContainment implements IssueRaiser {
+
+  private boolean raiseIssueOnMatch = true;
+
+
+  public ArgumentVerifierValueContainment(int position, @Nullable String name, Set<String> values) {
+    super(position, name, values);
+  }
+
+
+  public ArgumentVerifierValueContainment(int position, @Nullable String name, String value) {
+    this(position, name, SetUtils.immutableSetOf(value));
+  }
+
+  public ArgumentVerifierValueContainment(int position, String value, boolean raiseIssueOnMatch) {
+    this(position, null, SetUtils.immutableSetOf(value));
+    this.raiseIssueOnMatch = raiseIssueOnMatch;
+  }
+
+  public ArgumentVerifierValueContainment(int position, String name, String value, boolean raiseIssueOnMatch) {
+    this(position, name, SetUtils.immutableSetOf(value));
+    this.raiseIssueOnMatch = raiseIssueOnMatch;
+  }
+
+  public ArgumentVerifierValueContainment(int position, Set<String> values, boolean raiseIssueOnMatch) {
+    super(position, null, values);
+    this.raiseIssueOnMatch = raiseIssueOnMatch;
+  }
+
+  public ArgumentVerifierValueContainment(int position, String name, Set<String> values, boolean raiseIssueOnMatch) {
+    super(position, name, values);
+    this.raiseIssueOnMatch = raiseIssueOnMatch;
+  }
```

Comment:
For me, this is a typical indicator for a Builder class instead of so many constructors.